test/Directory.build.props

```diff
@@ -0,0 +1,42 @@
+<Project>
+  <PropertyGroup>
+    <DisableImplicitFrameworkReferences Condition=" '$(TargetFramework)' == 'net452' ">true</DisableImplicitFrameworkReferences>
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'netcoreapp1.1' ">
+    <DefineConstants>$(DefineConstants);CORE</DefineConstants>
+  </PropertyGroup>
+
+  <PropertyGroup Condition=" '$(Configuration)' == 'MonoRelease' ">
+    <DefineConstants>$(DefineConstants);MONO</DefineConstants>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <ProjectUnderTest>$(MSBuildProjectName.Substring(0, $(MSBuildProjectName.IndexOf(".Tests"))))</ProjectUnderTest>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="../../src/Nancy/Nancy.csproj" />
+    <ProjectReference Include="../../src/$(ProjectUnderTest)/$(ProjectUnderTest).csproj" />
+    <ProjectReference Include="../../src/Nancy.Testing/Nancy.Testing.csproj" />
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(MSBuildProjectName)' != 'Nancy.Tests' and '$(MSBuildProjectName)' != 'Nancy.ViewEngines.Razor.Tests.Models' ">

```

Comment:
This also a bit of a hack that we hopefully can clean up down the road. Basically we cannot include these files if we`re building `Nancy.Tests` because the files themselves are in that project, so we end up getting warnings about the files being declared twice.

If we include them for `Nancy.ViewEngines.Razor.Tests.Models` then we end up with a similar error because it will end up including the files both for `Nancy.ViewEngines.Razor.Tests.Models` and `Nancy.ViewEngines.Razor.Tests`